Source/stores.cpp

```diff
@@ -1658,6 +1658,7 @@ void SmithRepairItem()
 	}
 
 	myPlayer.InvList[i]._iDurability = myPlayer.InvList[i]._iMaxDur;
+	CalcPlrInv(myPlayer, true);

```

Comment:
I don't think this works as expected? There's an early return a few lines up if an equipped item is repaired, this would only run if a backpack item is repaired.